In [2]:
from movielens import MovieLens
import pandas as pd

In [3]:
lens = MovieLens("neo4j")

### Find out movies' name which contains "pacific" (Case in-sensitive) 

In [3]:
pd.DataFrame(lens.find_movies_by_name("pacific"))

movieId                                              title
0   25878                          Across the Pacific (1942)
1    3143                         Hell in the Pacific (1968)
2   90260              Hotel Pacific (Zaklete rewiry) (1975)
3    6435                           Operation Pacific (1951)
4    3219                             Pacific Heights (1990)
5  103228                                 Pacific Rim (2013)
6  129843  Plastic Paradise: The Great Pacific Garbage Pa...
7   87435  Sergeant Körmy and the South Pacific (Vääpeli ...
8    2941                               South Pacific (1958)
9  126430                                 The Pacific (2010)

### Find out movies' name which contains "Phenomenon" (Case sensitive) 

In [4]:
pd.DataFrame(lens.find_movies_by_name("Phenomenon", case_insensitive=True))

movieId              title
0     802  Phenomenon (1996)

# Find a movie with exact ID 
###  What is the movie name of which has movieId as 6000?

In [13]:
pd.DataFrame(lens.find_movie_by_id("296"))

average_rate  count movieId                title
0      4.174231  67310     296  Pulp Fiction (1994)

So average rate is 4.17, and total number of rating is 67,310. 

# Recommendation

### I like Pulp fiction (296), Phenomenon (802), What movie you can recommend it to me?

In [5]:
pd.DataFrame(lens.recommendation(['296', '802'], rating=4.5, limit=7))

average_rate  count movieId                                     title
0      4.576798  19727     318          Shawshank Redemption, The (1994)
1      4.519893  17343      50                Usual Suspects, The (1995)
2      4.599935  12253     858                     Godfather, The (1972)
3      4.562500      8  106503             Enemies of Reason, The (2007)
4      4.500000      7  101850  Death on the Staircase (Soupçons) (2004)
5      4.583333      6   26416                           Best Boy (1979)
6      4.800000      5   78064               Ween Live in Chicago (2004)

Based on users who give good rates to "Pulp finction" & "Phenomenon" (both starred by John Travolta), 
we can search other movies which they give generous score already. 

So *Shawshank Redemption*, *Usual Suspects*, *Godfather* is top 3 recommendation for those people, like me. 

### I like Pendorum (71135), Prometheus (94864), What **SF movie** you can recommend it to me?

In [4]:
pd.DataFrame(lens.recommendation(['94864', '71135'], rating=4.5, limit=7, category='Sci-Fi'))

average_rate  count movieId  \
0      4.586667    150    2571   
1      4.503597    139   79132   
2      4.553435    131    1214   
3      4.534091     44   27611   
4      4.709677     31    7099   
5      4.653846     26  109487   
6      4.541667     24   52885   

                                               title  
0                                 Matrix, The (1999)  
1                                   Inception (2010)  
2                                       Alien (1979)  
3                        Battlestar Galactica (2003)  
4  Nausicaä of the Valley of the Wind (Kaze no ta...  
5                                Interstellar (2014)  
6                          Paprika (Papurika) (2006)

As you see above, you only get the recommendation of Sci-fi movies, you can also put *Action*, or even *Romance*

In [6]:
pd.DataFrame(lens.recommendation(['94864', '71135'], rating=4.0, limit=7, category='Romance'))

average_rate  count movieId  \
0      4.182203    354     356   
1      4.231132    318   60069   
2      4.128866    291    7361   
3      4.029520    271    4995   
4      4.150407    246    1265   
5      4.113043    230    1704   
6      4.153302    212    4973   

                                               title  
0                                Forrest Gump (1994)  
1                                      WALL·E (2008)  
2       Eternal Sunshine of the Spotless Mind (2004)  
3                           Beautiful Mind, A (2001)  
4                               Groundhog Day (1993)  
5                           Good Will Hunting (1997)  
6  Amelie (Fabuleux destin d'Amélie Poulain, Le) ...

Rate is not that bigger than Sci-fi, but still...

I'm a big fan of Korean movie (of course, I'm coming from there) 

Especially, Joint Security Area(Thriller/Drama, 51709), Host(Scifi, 27397), Taeguki(Korean war, 27869). 

Then what could be my recommendation?

In [6]:
pd.DataFrame(lens.recommendation(['51709', '27397', '27869'], rating=4.5, limit=7))

average_rate  count movieId                                        title
0      4.534722     72   27773                               Old Boy (2003)
1      4.619048     42    2019  Seven Samurai (Shichinin no samurai) (1954)
2      4.585714     35    2288                            Thing, The (1982)
3      4.596154     26   31364   Memories of Murder (Salinui chueok) (2003)
4      4.557692     26    5291                   Rashomon (Rashômon) (1950)
5      4.522727     22    7371                              Dogville (2003)
6      4.533333     15     909                        Apartment, The (1960)